### PR TITLE
Bring back desync_finder from the dead.

### PR DESCRIPTION
**Pull request by tdaede**
Activate it with feature=desync_finder rather than debug.

Print to stdout rather than stderr because the latter deadlocks.

### CODE REVIEW

File: src/context.rs [hunk 1]
```diff
@@ -1866,22 +1866,23 @@ impl FieldMap {
     for (name, start, end) in &self.map {
       // eprintln!("{} {} {} val {}", name, start, end, addr);
       if addr >= *start && addr < *end {
-        eprintln!(" CDF {}", name);
-        eprintln!("");
+        println!(" CDF {}", name);
+        println!("");
```
Comment (lu-zero) [inline src/context.rs]: clippy is unhappy.

File: src/context.rs [hunk 2]
```diff
@@ -1866,22 +1866,23 @@ impl FieldMap {
...
 
 macro_rules! symbol_with_update {
   ($self:ident, $w:ident, $s:expr, $cdf:expr) => {
     $w.symbol_with_update($s, $cdf);
-    #[cfg(debug)] {
+    #[cfg(feature = "desync_finder")] {
+      //println!("{:?}", $cdf);
       if let Some(map) = $self.fc_map.as_ref() {
-        map.lookup($cdf.as_ptr() as usize);
+                map.lookup($cdf.as_ptr() as usize);
```
Comment (lu-zero) [inline src/context.rs]: while at it, this look strange.

File: src/context.rs [hunk 3]
```diff
@@ -3643,20 +3644,15 @@ impl<'a> ContextWriter<'a> {
...
-        0 => &mut self.fc.eob_flag_cdf16[plane_type][eob_multi_ctx],
-        1 => &mut self.fc.eob_flag_cdf32[plane_type][eob_multi_ctx],
-        2 => &mut self.fc.eob_flag_cdf64[plane_type][eob_multi_ctx],
-        3 => &mut self.fc.eob_flag_cdf128[plane_type][eob_multi_ctx],
-        4 => &mut self.fc.eob_flag_cdf256[plane_type][eob_multi_ctx],
-        5 => &mut self.fc.eob_flag_cdf512[plane_type][eob_multi_ctx],
-        _ => &mut self.fc.eob_flag_cdf1024[plane_type][eob_multi_ctx],
-      }
-    );
+    match eob_multi_size {
```
Comment (lu-zero) [inline src/context.rs]: maybe 
``` rust
let eob_flag_cdf = match eob_mult_size { ... };
symbol_with_update!(..., eob_flag_cdf);
```
works the same way?

Comment (tdaede) [reply src/context.rs]: Nope, as the cdf sizes are variable.

**Pull request merged**